File: ECommerce.Core/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;
using ECommerce.Core.Enums;

namespace ECommerce.Core.Entities;

public class Product : BaseEntity
{
    public string Headline { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Subtitle { get; set; }
    public string Sku { get; set; } = string.Empty; // Kept internally behind the scenes

    // Visibility
    public bool IsActive { get; set; } = true;
    public bool IsNew { get; set; } = false;

    // Pricing & Inventory natively replacing variants
    public decimal PurchaseRate { get; set; }
    public decimal Price { get; set; }
    public decimal? CompareAtPrice { get; set; }
    public int StockQuantity { get; set; }

    // Content fields matching frontend text areas
    public string? BenefitsTitle { get; set; }
    public string? BenefitsContent { get; set; }
    public string? UsageTitle { get; set; }
    public string? UsageContent { get; set; }
    public string? SideEffectsTitle { get; set; }
    public string? SideEffectsContent { get; set; }

    // Relationships
    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}

File: ECommarce.View/src/app/core/models/product.ts
export interface ProductImage {
  id: number;
  imageUrl: string;
  altText?: string;
  isPrimary: boolean;
}

export interface Product {
  id: number;
  name: string;
  headline: string;
  slug: string;
  subtitle?: string;
  imgUrl: string;
  images: ProductImage[];
  benefitsTitle?: string;
  benefitsContent?: string;
  sideEffectsTitle?: string;
  sideEffectsContent?: string;
  usageTitle?: string;
  usageContent?: string;
  reviewsTitle?: string;
  stockQuantity?: number;
  purchaseRate: number;
  price: number;
  compareAtPrice?: number;
  isActive: boolean;
  isNew: boolean;
  categoryId?: number;
  categoryName?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface RelatedProduct {
  id: number;
  name: string;
  headline: string;
  price: number;
  compareAtPrice?: number;
  imgUrl: string;
  slug: string;
}
